FHE/FHE_Keys.cpp

```diff
@@ -6,153 +6,202 @@
 
 #include "Math/modp.hpp"
 
+/**
+ * FFI Exports
+ */
 
-FHE_SK::FHE_SK(const FHE_PK& pk) : FHE_SK(pk.get_params(), pk.p())
+unique_ptr<FHE_KeyPair> new_keypair(const FHE_Params &params)
 {
+  auto res = unique_ptr<FHE_KeyPair>(new FHE_KeyPair(params));
+  res->generate();
+
+  return res;
+}
+
+unique_ptr<FHE_PK> get_pk(const FHE_KeyPair &pair)
+{
+  return unique_ptr<FHE_PK>(new FHE_PK(pair.pk));
 }
 
-FHE_SK::FHE_SK(const FHE_Params& pms) :
-    FHE_SK(pms, pms.get_plaintext_modulus())
+unique_ptr<FHE_SK> get_sk(const FHE_KeyPair &pair)
 {
+  return unique_ptr<FHE_SK>(new FHE_SK(pair.sk));
 }
 
+unique_ptr<Ciphertext> encrypt(const FHE_KeyPair &pair, const Plaintext_<FFT_Data> &mess)

```

Comment:
Will we always want to encrypt/decrypt through the whole keypair, or should we also expose the underlying `pk.encrypt` / `sk.decrypt` for FFI?

Reply:
see follow up